### DISCUSSION TITLE
Tips on debugging and handling errors?

### DISCUSSION THREAD

**Discussion by isConic** [leon-thomm/Ryven#99] [Q&A] [answered]
I was considering creating a try catch in `update_event` and an extra output node to send an error signal
![image](https://user-images.githubusercontent.com/48657113/158189438-63e96ce9-de7f-4542-95c8-d359384ce683.png)
  
How do you do it? Are there built-in Ryven features to help me debug? 
How do you catch errors? Sometimes something would break and the only way I would be able to tell that an error was being thrown is by observing wether or not my output changes as my inputs do.

**Comment by leon-thomm** [comment, marked as answer]
Ryven executes the `update_event` internally inside a `try` block already. On error, the traceback is printed if you enabled info messages (currently `Options -> Info Messages -> Enable` in Ryven, and it will print to the builtin console hidden at the very bottom of the window). You will soon also be able to direct all output to your terminal instead.

Of course, you can implement your own more optimized error handling inside your `update_event`. I recently also added an indicator for errors to the framework (suggestion from #94, commit: leon-thomm/ryvencore-qt@48bab818bfacd4a5364b8e27b178c7fa15a27534) which will be included in the next version soon.

**Reply by isConic** [reply]
That seems to work. Thanks!
